Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/Listener.cs

```diff
@@ -108,72 +108,91 @@ class Listener : IDisposable {
             await OnFileChanged(sender, e);
         }
         private async Task OnFileChanged(object sender, System.IO.FileSystemEventArgs e) {
-            // Skip directory change events and when rpc has ben disconnected
-            if (!e.IsDirectoryChanged() && _rpc != null) {
-                // Create something to match with
-                var item = new InMemoryDirectoryInfo(_root, new string[] { e.FullPath });
-
-                // See if this matches one of our patterns.
-                if (_matcher.Execute(item).HasMatches) {
-                    // Send out the event to the language server
-                    var renamedArgs = e as System.IO.RenamedEventArgs;
-                    var didChangeParams = new DidChangeWatchedFilesParams();
-
-                    // Visual Studio actually does a rename when saving. The rename is from a file ending with '~'
-                    if (renamedArgs == null || renamedArgs.OldFullPath.EndsWith("~")) {
-                        renamedArgs = null;
-                        didChangeParams.Changes = new FileEvent[] { new FileEvent() };
-                        didChangeParams.Changes[0].Uri = new Uri(e.FullPath);
-
-                        switch (e.ChangeType) {
-                            case WatcherChangeTypes.Created:
-                                didChangeParams.Changes[0].FileChangeType = FileChangeType.Created;
-                                break;
-                            case WatcherChangeTypes.Deleted:
-                                didChangeParams.Changes[0].FileChangeType = FileChangeType.Deleted;
-                                break;
-                            case WatcherChangeTypes.Changed:
-                                didChangeParams.Changes[0].FileChangeType = FileChangeType.Changed;
-                                break;
-                            case WatcherChangeTypes.Renamed:
-                                didChangeParams.Changes[0].FileChangeType = FileChangeType.Changed;
-                                break;
-
-                            default:
-                                didChangeParams.Changes = Array.Empty<FileEvent>();
-                                break;
-                        }
-                    } else {
-                        // file renamed
-                        var deleteEvent = new FileEvent();
-                        deleteEvent.FileChangeType = FileChangeType.Deleted;
-                        deleteEvent.Uri = new Uri(renamedArgs.OldFullPath);
 
-                        var createEvent = new FileEvent();
-                        createEvent.FileChangeType = FileChangeType.Created;
-                        createEvent.Uri = new Uri(renamedArgs.FullPath);
+            // Skip directory change events
```

Comment:
Ugh sorry about the formatting showing up as one big change. I should have put in the logic change as its own commit, then the early returns as another one so it would be easier to see the diff.

The only changes are:

1. Make sure the _rpc isn't null when you try to use it
2. Catch all ConnectionLostExceptions from _rpc calls and ignore them.